src/AmalgamatorSession.ts

```diff
@@ -637,12 +637,19 @@ export class AmalgamatorSession extends LoggingDebugSession {
         response: DebugProtocol.Response,
         args: any
     ): Promise<void> {
-        // XXX: Which childDap to send a customRequest to? Probably needs domain specific knowledge.
-        const childResponse = await this.childDaps[0].customRequest(
-            command,
-            args
-        );
-        response.body = childResponse.body;
-        this.sendResponse(response);
+        if (command === 'cdt-gdb-adapter/Memory') {
```

Comment:
Please do type checking when handling any types, see[ this bit ](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/blob/16c45b9e1b5f32e29d330dcd6dcc8ba7416c7b81/src/GDBDebugSession.ts#L1249-L1270)of cdt-gdb-adapter for an example.

Reply:
I have updated it